controllers/suite_test.go

```diff
@@ -251,19 +253,23 @@ var _ = BeforeSuite(func() {
 
 	Expect(err).NotTo(HaveOccurred())
 
+	CreateNamespaceWithContext(ctx, &appNamespace)
+
 	go func() {
 		defer GinkgoRecover()
 		err = mgr.Start(ctrl.SetupSignalHandler())
 		Expect(err).ToNot(HaveOccurred())
 	}()
 
+	ApplyKuadrantCR(appNamespace)
```

Comment:
Won't having this here also affect the tests that need to create and delete Kuadrant CRs, such as the ones for `KuadrantReconciler` (if we had any 😕)

I liked the approach of having this in the `BeforeAll` and then deleting the CR in `AfterAll` funcs added to the DNS controller tests. WDYT?

Reply:
Yeah, we can do this is the `BeforeAll` /`AfterAll` funcs instead. Only thing with this is we'd need to have the tests in an `Ordered` manner - https://onsi.github.io/ginkgo/#setup-in-ordered-containers-beforeall-and-afterall so we'll lose running the tests in a random order, which may not be ideal either on making sure there's no dependency / leak between tests 🤔 If we are good with this, I can do this in the `BeforeAll/AfterAll` instead

Reply:
My 2¢ – I guess "ordered" is not as much of an issue as "sequential" is, and we already have sequential tests. So I suppose _random_ is adding just a marginal unit of chance for writing better tests. Not being able to run tests _concurrently_ is our biggest constraint IMO.

Reply:
Yeah, that's true, I'll push this change to use  do this in `BeforeAll/AfterAll` instead

Reply:
Do you have to do the `BeforeAll` with the `AfterAll```? What I am thinking is if there is a `BeforeEach` that checks if kuadrant is installed and if it is missing install it. Then after the all specs have ran you run the `AfterAll`. Not sure if this would get away from needing the ordered flag but in theory it should still allow us to run tests concurrently. "Maybe"

Reply:
No, I don't think you have to the `BeforeAll/AfterAll` but if you use any one of them, them it needs to marked as Ordered. Using `BeforeEach`doesn't sound like a nice solution to me 🤔 I guess the dificulty is that most tests require a kuadrant instance, some doesn't matter, some does not want an install, and some ideally wants to set up the kuadrant instance in their own testing scenarios 🤔 This is an intermediate solution anyway to just speed things up a good margin still while we look into getting the tests to run concurrently in the future